sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/PersistentStorage/StorageHelper.cs

```diff
@@ -43,12 +44,11 @@ internal static string GetStorageDirectory(IPlatform platform, string? configure
                     || TryCreateTelemetryDirectory(platform: platform, path: "/var/tmp/", createdDirectoryPath: out dirPath)
                     || TryCreateTelemetryDirectory(platform: platform, path: "/tmp/", createdDirectoryPath: out dirPath))
                 {
-                    s_defaultStorageDirectory = dirPath;
-                    return s_defaultStorageDirectory;
+                    return dirPath;
                 }
             }
 
-            return s_defaultStorageDirectory;
+            throw new InvalidOperationException("Unable to determine default storage directory.");
```

Comment:
Why is this InvalidOperation?

Reply:
InvalidOperation is the default exception that's being used throughout most of the SDK.
We have a work item to fully review all exceptions and logging before GA.

Reply:
Why are we changing the behavior of this method to throw?

Reply:
> Why are we changing the behavior of this method to throw?

This method seemed the more appropriate location to throw. 
Previously, this method returned a null, and the calling method threw.

Reply:
My recommendation is to restore the previous behavior. `InvalidOperationException` is not a better fit here and we don't need to throw if storage directory is null, rather we could log to event source.